### PR TITLE
Feature/k 870 implement the endpoint for searching for exact availability

### PR DESCRIPTION
**Pull request by AlexeyShok**
Implemented endpoint for searching for exact availability

### CODE REVIEW

File: HappyTravel.Komoro.Api/Services/Availabilities/AccommodationAvailabilities/AccommodationAvailabilityService.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+public class AccommodationAvailabilityService : IAccommodationAvailabilityService
+{
+    public AccommodationAvailabilityService(KomoroContext komoroContext)
+    {
+        _komoroContext = komoroContext;
+    }
+
+
+    public async Task<Result<AccommodationAvailability>> Get(string supplierCode, string propertyCode, string availabilityId,
```
Comment (oleg-n-k) [inline HappyTravel.Komoro.Api/Services/Availabilities/AccommodationAvailabilities/AccommodationAvailabilityService.cs:20]: Looks like the method is a bit overloaded.
It's better to separate the logic with additional methods.
Like GetRooms, GetInventories etc

File: HappyTravel.Komoro.Api/Services/Availabilities/RoomContractSetAvailabilities/RoomContractSetAvailabilityService.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+public class RoomContractSetAvailabilityService : IRoomContractSetAvailabilityService
+{
+    public RoomContractSetAvailabilityService(KomoroContext komoroContext)
+    {
+        _komoroContext = komoroContext;
+    }
+
+
+    public async Task<Result<RoomContractSetAvailability?>> Get(string supplierCode, string propertyCode, string availabilityId,
```
Comment (oleg-n-k) [inline HappyTravel.Komoro.Api/Services/Availabilities/RoomContractSetAvailabilities/RoomContractSetAvailabilityService.cs:20]: The method is too long

File: HappyTravel.Komoro.Api/Services/Availabilities/AccommodationAvailabilities/AccommodationAvailabilityService.cs (hunk 1 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.Api/Services/Availabilities/AccommodationAvailabilities/AccommodationAvailabilityService.cs:20]: Fixed in https://github.com/happy-travel/komoro/pull/144

File: HappyTravel.Komoro.Api/Services/Availabilities/RoomContractSetAvailabilities/RoomContractSetAvailabilityService.cs (hunk 2 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.Api/Services/Availabilities/RoomContractSetAvailabilities/RoomContractSetAvailabilityService.cs:20]: Fixed in https://github.com/happy-travel/komoro/pull/144

**Pull request merged**